src/components/common/FileUploadPreview.vue

```diff
@@ -137,9 +146,13 @@ export default class FileUploadPreview extends Mixins(DocumentMixin) {
     }
   }
 
-  /** Sets custom error message if is changed from its initial value. */
+  /**
+   * Automatically sets custom error message if prop changed, or if called, and syncs value back
+   * to the parent component.
+   */
   @Watch('customErrorMessage')
-  private onCustomErrorMessage (val: string): void {
+  @Emit('update:customErrorMessage')
+  private setCustomErrorMessage (val: string): void {
```

Comment:
This updates the parent component (via `sync`, if used), so that the parent can set a new value if needed. This allows setting errors above or from the parent.

(There's actually a 3rd place that errors are generated: the rules. But what I've done above is fairly clean and handles all cases.)